dcos/http/client/client.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2016 Mesosphere, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import "net/http"
+
+// NewClient returns a client with CAPool when passed a CA certificate path and
+// a Authorization header when passed a IAM configuration path. If the path to CA
+// certificate or IAM config is an empty string then the corrosponding client
+// configurations which they set are ignored.
+func NewClient(caCertificatePath string, iamConfigPath string) (*http.Client, error) {
```

Comment:
can we make it to accept functional parameters here? i assume we may want to extend NewClient in future with DC/OS specific options. This will ensure interface isn't broken for current users.